.github/workflows/pr_benchmarks.yml

```diff
@@ -0,0 +1,71 @@
+name: pr-benchmarks
```

Comment:
Why do we have a new workflow instead of reusing the previous one? All PR workflows must also have an associated non-PR workflow btw.

Reply:
benchmarks cannot work with `pytest-xdist`. I can add a new step to an existing workflow though. I can updated the existing integration test workflows.